mango-python/bdgenomics/mango/variants.py

```diff
@@ -76,3 +83,165 @@ def viewPileup(self, contig, start, end, reference = 'hg19', label = "Variants",
             locus="%s:%i-%i" % (contig, start, end)
             return pileup.PileupViewer(locus=locus, reference=reference, tracks=tracks)
 
+
+class VariantsPerSampleDistribution(object):
+    """ VariantsPerSampleDistribution class.
+    VariantsPerSampleDistribution computes a distribution of the count of variants per sample.
+    """
+    def __init__(self,ss,genotypeRDD):
+        """
+        Initializes a VariantsPerSampleDistribution class.
+        Computes the counts of variants per sample from a genotypeRDD.
+
+        :param ss: SparkContext
+        :param genotypeRDD: bdgenomics genotypeRDD
+        """
+        df =  genotypeRDD.toDF()
+        data =  df.where( (df.alleles[0] == u'ALT') | (df.alleles[1] == u'ALT')).groupBy("sampleid").count().collect()
+        self.counts = [ sample.asDict()['count'] for sample in data ]
+
+    def plot(self, testMode = False, **kwargs):
+        """
+        Plots final distribution values and returns the plotted distribution as a list
+
+        Returns:
+          matplotlib axis to plot and computed data
+        """
+
+        if(not testMode):
+            figsize = kwargs.get('figsize',(10,5))
+            fig, ax = plt.subplots(figsize=figsize)
+            bins = kwargs.get('bins',100)
+            hist = ax.hist(self.counts, bins=bins)
+            return ax, self.counts
+        else:
+            return None, self.counts
+
+
+class HetHomRatioDistribution(object):
+    """ HetHomRatioDistribution class.
+    HetHomRatioDistribution computes a distribution of (Heterozygote/Homozygote) ratios from a genotypeRDD.
+    """
+    def __init__(self,ss,genotypeRDD):
+        """
+        Initializes HetHomRatioDistribution class.
+        Retrieves the call rate and missing rate for each sample in a genotypeRDD
+
+        Args:
+            :param ss: SparkContext
+            :param genotypeRDD: bdgenomics genotypeRDD
+        """
+
+        df =  genotypeRDD.toDF()
+        data_het =  df.where( (df.alleles[0] == u'REF') & (df.alleles[1] == u'ALT')).groupBy("sampleid").count().collect()
+        self.hets =  { sample.asDict()['sampleid']:sample.asDict()['count'] for sample in data_het }
+        data_hom =  df.where( (df.alleles[0] == u'ALT') & (df.alleles[1] == u'ALT')).groupBy("sampleid").count().collect()
```

Comment:
Maybe we could write this more clever to only do 1 collect. It will take half as much time. You could do a map to hom/het and then reducebykey